### PR TITLE
added `withVariant` documentation block

### LINKED ISSUE #4722: Document `withVariant`.
**Is your feature request related to a problem? Please describe.**
[Recently](https://github.com/bbc/simorgh/pull/4567), a `withVariant` HOC was added to simorgh. We need to update the [documentation](https://github.com/bbc/simorgh/#a-page-render-lifecycle) on this.

**Describe the solution you'd like**
Update the [readme](https://github.com/bbc/simorgh/#a-page-render-lifecycle) to contain `withVariant`.

### PR DESCRIPTION
**Pull request by itimirichard**
Resolves #4722

**Overall change:** _Added a documentation block for `withVariant` HOC._

**Code changes:**

- _Updated README.md ._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,12 +67,20 @@ This process is repeated for each block within an article, ultimately rendering
...
 - withPageWrapper
 - withLoading
 - withError
 - withData
 
+#### withVariant
+
+The variant HOC ensures that services that have variants (e.g. `simp`, `lat`) always redirects to a url that renders the appropriate variant.
+
+- If a user navigates to a url without providing the variant, and variant is set in cookie, the cookie variant page is rendered. Otherwise, the default variant page is rendered
```
Comment (samora) [inline README.md]: Let’s keep the format of the document by using paragraphs instead of a list.

File: README.md [hunk 2]
```diff
@@ -67,12 +67,20 @@ This process is repeated for each block within an article, ultimately rendering
...
+- If a user navigates to a url with a variant, and variant is set in cookie, the cookie variant page is rendered. Otherwise, the requested variant page is rendered.
```
Comment (samora) [inline README.md]: If the user navigates to a url with a provided variant do we really redirect them to the cookie variant regardless?

**Review by samora — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (itimirichard) [reply README.md]: Yes I believe so. https://github.com/bbc/simorgh/files/3792991/Simorgh.variant.User.Journeys.1.pdf

File: README.md (hunk 1 above)
Comment (itimirichard) [reply README.md]: okay. fixed!

**Review by andrewscfc — APPROVED** [review]
Just looked through the `PageHandler` these docs match what I saw :)

**Review by samora — APPROVED** [review]

**Review by chris-hinds — APPROVED** [review]

**Pull request merged**